Add Qwen2 Models

jiangsier-xyz opened the pull request:
Add newest open-sourced qwen2 models, which achieved amazing evaluation results!

See: [Hello Qwen2](https://qwenlm.github.io/blog/qwen2/)

And:
- Remove duplicate tests.
- Refactor a small amount of code.
- Fix #1267 

<!-- Thank you so much for your contribution! -->
<!-- Please fill in all the sections below. -->

<!-- Please open the PR as a draft initially. Once it is reviewed and approved, we will ask you to add documentation and examples. -->
<!-- Please note that PRs with breaking changes will be rejected. -->
<!-- Please note that PRs without tests will be rejected. -->

<!-- Please note that PRs will be reviewed based on the priority of the issues they address. -->
<!-- We ask for your patience. We are doing our best to review your PR as quickly as possible. -->
<!-- Please refrain from pinging and asking when it will be reviewed. Thank you for understanding! -->


## Issue
<!-- Please paste the link to the issue this PR is addressing. For example: https://github.com/langchain4j/langchain4j/issues/1012 -->


## Change
<!-- Please describe the changes you made. -->


## General checklist
<!-- Please double-check the following points and mark them like this: [X] -->
- [x] There are no breaking changes
- [x] I have added unit and integration tests for my change
- [x] I have manually run all the unit and integration tests in the module I have added/changed, and they are all green

geoand commented:
Hi,

I think it would be better if one commit adds the new models and one does the refactoring because it would make it easier to review.

jiangsier-xyz committed:
Add Qwen2 Models

Add newest open-sourced qwen2 models, which achieved amazing evaluation results!

https://qwenlm.github.io/blog/qwen2/

jiangsier-xyz committed:
Remove Duplicate Test

jiangsier-xyz committed:
Refactor Some Codes

jiangsier-xyz commented:
> Hi,
> 
> I think it would be better if one commit adds the new models and one does the refactoring because it would make it easier to review.

OK.

jiangsier-xyz committed:
Fix #1267

geoand reviewed (approved):
LGTM
